### ISSUE TITLE
Feature Suggestion: There needs to be a way to "pin" and "unpin" items in the "Recent Items" list

### ISSUE THREAD

**Issue by yannduran** [luminous-software/start-page-plus#21]
![image](https://user-images.githubusercontent.com/2031126/65937674-f7b15500-e463-11e9-9238-18899eb2146e.png)

**Comment by Dean-NC** [comment]
Is this coming later or is it in now? I have 0.14.3 and when I hover a recent project, I see a square to the far right where the pin icon would be, but clicking it does nothing.

![Untitled](https://user-images.githubusercontent.com/12361341/66504476-25ffe780-ea97-11e9-8524-51284a1f1101.png)

It's like this for all IDE themes.

**Comment by yannduran** [comment]
Hi Dean! I'm so glad you like _Start Page+_

Yes, what you're seeing in an incomplete feature. I released a preview early to get a decent _Blue_ theme out to those who could use it as it was. _Start Page+_ is still in early preview, so there's still a number of things to add, tweak, and polish.

I love what you've added to your main menu. I see you have the _Start Page+_ project open, did you modify the project to add your menu modifications, or did you just customise the menu bar in Visual Studio? I ask because I accept pull requests for enhancements, or new features.

**Comment by Dean-NC** [comment]
OK, that explains the pinning...no worry...I fully understand early preview, and I'm in no hurry to get it.

I didn't modify the StartPage+ project, and I didn't even have it open in that picture; I just customized the menu bar. I didn't even notice that having StartPage+ docked caused the VS main window title to have the words "Start Page" in it...good catch...don't know if you want that or not.

I know you already know how to customize the VS main menu, but for others' sake, here's the steps: first right-click anywhere in the main menu bar (File, View, etc.), then choose "Customize". Then on the "Commands" tab, click "Add Command" button. In the Categories list, select "View", then wait for the Commands list to populate, then select the "ViewStartPagePlus" command. After adding the command to the menu, click the "Modify Selection" button, which allows you to rename the menu text (I used "Start"), and to use either the icon only, or as I did...the icon and text.

![VSCustStartMenu1](https://user-images.githubusercontent.com/12361341/66579999-b0098800-eb4b-11e9-96f1-9d41bac8a954.png)

![VSCustStartMenu2](https://user-images.githubusercontent.com/12361341/66580009-b39d0f00-eb4b-11e9-9785-aed569386c4f.png)

**Comment by yannduran** [comment]
Thanks for explaining that for people.

Here's what I did straight after your first comment:

![image](https://user-images.githubusercontent.com/2031126/66611228-93cc2080-ec09-11e9-88ea-4f279be95a6f.png)

**Comment by yannduran** [comment]
Hmm, interesting. I found that when you have no solution open, VS will display the name of the **_currently selected tab_** in the document well.

First off, in case you're wondering why my tabs look like they do, they're styled using [Tabs Studio](https://marketplace.visualstudio.com/items?itemName=SergeyVlasov.TabsStudio-documenttabsmanagerforVisualStudioIDE) (a paid extension), and I use _vertical tabs_ (also from Tabs Studio). And I have the _Output_ window docked in the document well. .

You can see in each of the screenshots that I have no solution open.

Here's the docked _Output_ window selected:

![image](https://user-images.githubusercontent.com/2031126/66612028-181fa300-ec0c-11e9-8740-361549041ebe.png)

Here's an Azure DevOps page selected:

![image](https://user-images.githubusercontent.com/2031126/66612162-a09e4380-ec0c-11e9-9232-5953b58b5a6a.png)

Here's a docked _Git History_ window:

![image](https://user-images.githubusercontent.com/2031126/66612291-0a1e5200-ec0d-11e9-9b49-73bfd1aef6fe.png)

And finally, here's the docked Start Page+ window selected:

![image](https://user-images.githubusercontent.com/2031126/66612324-2b7f3e00-ec0d-11e9-9ae0-b9e78bdcdc9f.png)

So it's not actually anything to do with Start Page+ itself, apart from the fact that it's a document tab.

**Comment by Dean-NC** [comment]
OK, that makes sense, thanks for letting us know! It's part of that ever-learning we have for Visual Studio.

**Comment by BOBJohnson** [comment]
So I have some Pinned items, and for the life of me, I couldn't figure out why I couldn't add a new one. Came here and found that it's not implemented yet. But wait, how do I have pinned items?

Well I found a workaround:
File -> Start Window
In the Start Window pin the project here. Close the start window.

Close and reopen the Start Page+. You now have a new pinned project (this should work for unpinning as well).

Thought I'd share this workaround in case anyone else happened across this issue thread.
